### ISSUE TITLE
Gamut question

### ISSUE THREAD

**Issue by virtualritz** [scrtwpns/pigment-mixing#2]
I apologize in advance if I am missing something obvious from your paper/talk with below questions.

TLDR; Why did you use `sRGB` and not `ACEScg` as the space to assume input/output RGB triplets for your algorithm to lie in?

In your paper you say:

> Since the mixtures of 𝒫* lie outside the RGB gamut [...]

What RGB gamut are you talking about here? `sRGB`’s? Any RGB space’s gamut?

I am inclined to read this as:

> Since (some of) the mixtures of 𝒫* lie outside (any) ~the~ RGB gamut [...]

Is this the correct way to read this? And if so: why not choose an RGB space with a larger gamut to do all the calculations in?

Modern displays – especially those used by CG professionals –, have much larger gamuts than that covered by `sRGB`. Could you elaborate why you choose `sRGB` at all for this research?

See e.g. [Wikipedia](https://en.wikipedia.org/wiki/DCI-P3) for an (incomplete) list of devices that have `P3 wide color gamut` nowadays.

To be practical: I would think that colors which lie outside of even [`ACEScg`](https://en.wikipedia.org/wiki/Academy_Color_Encoding_System#ACEScg) or [`ACES2065-1`](https://en.wikipedia.org/wiki/Academy_Color_Encoding_System#ACES2065-1) gamuts, e.g. something like *ultramarine*, would look much closer to their physical pigment counterpart, on a modern screen, if you did the maths in one of those spaces instead of `sRGB`.

**Comment by scrtwpns** [comment]
The whole point of Mixbox is to provide pigment-based color mixing in a way that is backwards compatible with all the legacy code out there. If someone was making new painting software today, they wouldn't need Mixbox to achieve accurate color mixing. Instead, they could just go ahead and do the right thing from start by making all the calculations in spectrum and only convert to device gamut for display. 

We created Mixbox to bring accurate pigment mixing back to software that was not written with spectral representation in mind. Since most software is written around the 3-channel RGB representation, we need to work with the gamut of "lowest common denominator", and that is the sRGB. In fact, it is not uncommon to see software that doesn't perform any kind of color management at all. In situation like this, when "unmanaged" RGB triplets get passed around, the most sensible thing to do is to interpret them as sRGB values. This is why we choose sRGB, to maximize compatibility with existing software.

However, the important thing to point out is that we only use sRGB at the interface, for input and output. The internal calculations are all preformed in full spectrum. This gets baked into the lookup table.

It is great to see those P3 capable devices getting more common, and we do plan to provide P3 compatible version of Mixbox in future.

**Comment by virtualritz** [comment]
> However, the important thing to point out is that we only use sRGB at the interface, for input and output. The internal calculations are all performed in full spectrum. This gets baked into the lookup table.

I presume interface is the `coefs` table? Could you publish the code to generate this table so someone could make a table for RGB inputs specified in spaces w. bigger gamuts, e.g. `ACEScg`?

Also: is the `sRGB` *2.2 gamma* removal baked into `coefs` (see #1 )?

To your point: you are still limited to the gamut of `sRGB` at the interface which, I agree, is a good assumption for `u8`. But a rather limiting one for `f32`.

I.e. if you have a color represented in your app as an `f32` RGB tuple in e.g.`ACEScg` gamut you will loose quite a bit of "possible input" at the extremes of the primaries.

The color has to be converted from `ACEScg` to `sRGB`. Colors close to the input's gamut's edges/corners will simply be clipped.

And this can not be compensated internally since the information is lost already before input – at the interface.
